lib/widgets/components/scaffold.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:coffeecard/base/strings_environment.dart';
+import 'package:coffeecard/base/style/colors.dart';
+import 'package:coffeecard/base/style/text_styles.dart';
+import 'package:coffeecard/cubits/environment/environment_cubit.dart';
+import 'package:coffeecard/widgets/components/dialog.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:gap/gap.dart';
+
+class AppScaffold extends StatelessWidget {
+  final Widget? title;
+  final Widget body;
+  final bool resizeToAvoidBottomInset;
+
+  AppScaffold({
+    required String title,
+    this.resizeToAvoidBottomInset = true,
+    required this.body,
+  }) : title = Text(title, style: AppTextStyle.pageTitle);
+
+  const AppScaffold.noTitle({
+    this.resizeToAvoidBottomInset = true,
+    required this.body,
+  }) : title = null;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomInset: resizeToAvoidBottomInset,
+      // The background color is set to avoid a thin line
+      // between the AppBar and the _EnvironmentBanner.
+      // The actual background of the body is defined
+      // in the child of the Expanded widget below.
+      backgroundColor: AppColor.primary,
+      // If a title is not provided, then the AppBar is an empty 24dp tall padding.
+      // Otherwise, display a normal AppBar with a title widget.
+      appBar: AppBar(
+        title: title,
+        elevation: 0,
+        toolbarHeight: (title == null) ? 24 : null,
+      ),
+      body: BlocBuilder<EnvironmentCubit, Environment>(
+        builder: (context, state) {
+          return Column(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              if (state.isTest) const _EnvironmentBanner(tappable: true),
+              Expanded(
+                child: Container(
+                  color: (title != null) ? AppColor.background : null,
```

Comment:
The new constructor `Scaffold.noTitle` does not imply that the background of the body will change.
The name `Scaffold.login` fits better in this scenario

Reply:
i see, i still don't agree with having a constructor fitted to a specific page. I propose to add an optional background field in the noTitle constructor to preserve the reuseable aspect, this way the loginPage can control the color of the background. Thoughts?

Reply:
Check the new changes